src/main/java/uk/gov/companieshouse/search/api/elasticsearch/AlphabeticalSearchRequests.java

```diff
@@ -0,0 +1,129 @@
+package uk.gov.companieshouse.search.api.elasticsearch;
+
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.sort.SortOrder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+import uk.gov.companieshouse.search.api.service.rest.RestClientService;
+
+import java.io.IOException;
+
+import static uk.gov.companieshouse.search.api.SearchApiApplication.APPLICATION_NAME_SPACE;
+
+@Component
+public class AlphabeticalSearchRequests {
+
+    @Autowired
+    private RestClientService searchRestClient;
+
+    @Autowired
+    private EnvironmentReader environmentReader;
+
+    @Autowired
+    private AlphabeticalSearchQueries alphabeticalSearchQueries;
+
+    private static final String INDEX = "ALPHABETICAL_SEARCH_INDEX";
+    private static final String RESULTS_SIZE = "ALPHABETICAL_SEARCH_RESULT_MAX";
+    private static final String ORDERED_ALPHA_KEY_WITH_ID = "ordered_alpha_key_with_id";
+    private static final Logger LOG = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    public SearchHits getBestMatchResponse(String orderedAlphakey, String requestId) throws IOException {
+        SearchRequest searchRequestBestMatch = createBaseSearchRequest(requestId);
+        searchRequestBestMatch.source(bestMatchSourceBuilder(
+            alphabeticalSearchQueries.createOrderedAlphaKeySearchQuery(orderedAlphakey),
+            ORDERED_ALPHA_KEY_WITH_ID, SortOrder.ASC));
+
+        SearchResponse searchResponse = searchRestClient.searchRestClient(searchRequestBestMatch);
+        return searchResponse.getHits();
+    }
+
+    public SearchHits getStartsWithResponse(String orderedAlphakey, String requestId) throws IOException {
+        LOG.info("A hit was not found for: " + orderedAlphakey + ", falling back to prefix on alphakey");
+        SearchRequest searchRequestStartsWith = createBaseSearchRequest(requestId);
+
+        searchRequestStartsWith.source(bestMatchSourceBuilder(
+            alphabeticalSearchQueries.createOrderedAlphaKeyKeywordQuery(orderedAlphakey),
+            ORDERED_ALPHA_KEY_WITH_ID, SortOrder.ASC));
+
+        SearchResponse searchResponse = searchRestClient.searchRestClient(searchRequestStartsWith);
+        return searchResponse.getHits();
+    }
+
+    public SearchHits getCorporateNameStartsWithResponse(String orderedAlphakey,
+        String requestId) throws IOException {
+
+        LOG.info("A hit was not found for: " + orderedAlphakey + ", falling back to corporate name");
```

Comment:
Is `A hit was not found for...` always the reason why this method is executed?

Reply:
In this case yes. This method is only ever executed if the previous search requests have returned no hits. Below is a little idea of when this method would be executed. 

1) Try to find an exact match
2) No exact match found, find the next best thing
3) (This would be the method highlighted above) Still no match - query against a different field in the index